ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+  private OgcFeatureCollectionTable ogcFeatureCollectionTable; // keep loadable in scope to avoid garbage collection
+
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with the topographic basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
+
+      // set the map to the mapview
+      mapView.setMap(map);
+
+      // prepare UI for interaction
+      populateUiWithCqlQueriesAndDateValues();
+
+      // define strings for the service URL and collection id
+      // note that the service defines the collection id which can be accessed via OgcFeatureCollectionInfo.getCollectionId().
+      String serviceUrl = "https://demo.ldproxy.net/daraa";
+      String collectionId = "TransportationGroundCrv";
+
+      // create an OGC feature collection table from the service url and collection id
+      ogcFeatureCollectionTable = new OgcFeatureCollectionTable(serviceUrl, collectionId);
+
+      // set the feature request mode to manual
+      // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+      ogcFeatureCollectionTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table
+      ogcFeatureCollectionTable.loadAsync();
+
+      // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+      ogcFeatureCollectionTable.addDoneLoadingListener(() -> {
+        if (ogcFeatureCollectionTable.getLoadStatus() == LoadStatus.LOADED) {
+
+          // create a feature layer and set a renderer to it to visualize the OGC API features
+          var featureLayer = new FeatureLayer(ogcFeatureCollectionTable);
+          var simpleRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.DARKMAGENTA), 3));
+          featureLayer.setRenderer(simpleRenderer);
+
+          // add the layer to the map
+          map.getOperationalLayers().add(featureLayer);
+
+          datasetExtent = ogcFeatureCollectionTable.getExtent();
+
+          // display result of query on the map
+          setInitialQueryOnOgcFeatureTable();
+
+        } else {
+          // show an alert dialog if there is a loading failure
+          new Alert(Alert.AlertType.ERROR, "Failed to load OGC Feature Collection Table: " +
+            ogcFeatureCollectionTable.getLoadError().getCause().getMessage()).show();
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Gets the total extent of the OGC feature collection table, and uses a basic query parameters to return 3000 features
```

Comment:
>Gets the total extent of the OGC feature collection table

Is that correct? Didn't you get the extent earlier when the table was loaded before you called this method?

Reply:
good spot: updated the comment